### PR TITLE
Fix 404 errors due to gh-pages

### PR DESCRIPTION
**Pull request by willrushi**
Fixed the issue where 404s would be returned when navigating to site

This issue was due to gh-pages not playing nicely with SPAs
https://create-react-app.dev/docs/deployment/#notes-on-client-side-routing

Fixed by adding a 404 page with some redirect magic as explained on above link

Reference tweet:
https://twitter.com/osquery/status/1397619077707964418

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -19,11 +20,35 @@
...
+<body>
     <noscript>
       You need to enable JavaScript to run this app.
     </noscript>
     <div id="root"></div>
-  </body>
-</html>
+</body>
+
+</html>
```
Comment (mikermcneil) [inline public/index.html:54]: nit: newline before eof for consistency
image

File: package.json [hunk 2]
```diff
@@ -1,136 +1,136 @@
...
+    },
+    "babel": {
+        "presets": [
+            "react-app"
+        ]
+    },
+    "eslintConfig": {
+        "extends": "react-app"
+    }
+}
```
Comment (willrushi) [inline package.json:136]:
```suggestion
}

```

File: public/404.html [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            (l.search ? '&' + l.search.slice(1).replace(/&/g, '~and~') : '') +
+            l.hash
+        );
+    </script>
+</head>
+
+<body>
+</body>
+
+</html>
```
Comment (willrushi) [inline public/404.html:42]:
```suggestion
</html>

```

File: public/index.html (hunk 1 above)
Comment (willrushi) [inline public/index.html:54]:
```suggestion
</html>

```

**Review by willrushi — COMMENTED** [review]
Newlines added

File: public/index.html (hunk 1 above)
Comment (willrushi) [reply public/index.html]: Cheers, fixed 👍

**Comment by theopolis** [comment]
I tested this locally and it all of the functionally still works, happy to merge and deploy.

**Review by directionless — APPROVED** [review]
Seem fine. Feels complex. But I'm not a js person

**Comment by mikermcneil** [comment]
🙌 Thanks everybody, and nice work @willrushi!

**Pull request merged**